### PR TITLE
Index endpoint details hash

### PR DESCRIPTION
**Pull request by Rosa-Fox**
https://trello.com/c/pegK7Wtw/485-index-endpoint-should-return-details-hash

### CODE REVIEW

File: app/queries/get_content_collection.rb [hunk 1]
```diff
@@ -11,14 +11,13 @@ def initialize(content_format:, fields:, publishing_app: nil, locale: nil)
 
     def call
       validate_fields!
-
       content_items.map do |content_item|
         hash = Presenters::Queries::ContentItemPresenter.new(
```
Comment (tuzz) [inline app/queries/get_content_collection.rb]: I think we should rename the `hash` variable to `presenter`.

Alternatively, we could call chain a call to `#present` onto the presenter, e.g.

```ruby
hash = Presenters::Queries::ContentItemPresenter.new( ... ).present
```

File: app/queries/get_content_collection.rb [hunk 2]
```diff
@@ -87,5 +86,9 @@ def live_version(item)
         @live_versions[item.id]
       end
     end
+
+    def select_output_fields_only(hash)
+      hash.present.as_json.select { |k, v| output_fields.map(&:to_s).include?(k) }
+    end
```
Comment (tuzz) [inline app/queries/get_content_collection.rb]: You can probably use [Hash#slice](http://api.rubyonrails.org/classes/Hash.html#method-i-slice) for this. You won't need to `#to_s` the fields if you slice before calling `#as_json`.

File: spec/queries/get_content_collection_spec.rb [hunk 3]
```diff
@@ -134,4 +134,18 @@
...
+        expect(Queries::GetContentCollection.new(
+          content_format: 'topic',
+          fields: ['details'],
+          publishing_app: 'publisher'
+        ).call).to eq([
+          { "details" => {"foo" => "bar"}, "publication_state" => "draft" },
+          { "details" => {"baz" => "bat"}, "publication_state" => "draft" }
+        ])
+    end
+  end
```
Comment (tuzz) [inline spec/queries/get_content_collection_spec.rb]: :+1:

**Comment by Rosa-Fox** [comment]
@tuzz thanks for the feedback :) I have altered. Let me know if you think there should be any further changes.

**Comment by tuzz** [comment]
:+1: :rocket:

**Pull request merged**